src/PostgREST/Auth.hs

```diff
@@ -20,33 +20,40 @@ module PostgREST.Auth
 import qualified Crypto.JWT          as JWT
 import qualified Data.Aeson          as JSON
 import qualified Data.HashMap.Strict as M
+import qualified Data.Text           as T
 import qualified Data.Vector         as V
 
-import Control.Lens            (set)
-import Control.Monad.Except    (liftEither)
-import Data.Either.Combinators (mapLeft)
-import Data.Time.Clock         (UTCTime)
+import Control.Lens              (set)
+import Control.Monad.Except      (liftEither)
+import Data.Either.Combinators   (mapLeft)
+import Data.List                 (lookup)
+import Data.Time.Clock           (UTCTime)
+import Network.HTTP.Types.Header (hAuthorization)
+import Network.Wai               (Request (..))
 
 import PostgREST.Config (AppConfig (..), JSPath, JSPathExp (..))
 import PostgREST.Error  (Error (..))
 
-import Protolude
-
+import Protolude      hiding (toS)
+import Protolude.Conv (toS)
 
 type JWTClaims = M.HashMap Text JSON.Value
 
 -- | Receives the JWT secret and audience (from config) and a JWT and returns a
 -- map of JWT claims.
 jwtClaims :: Monad m =>
-  AppConfig -> LByteString -> UTCTime -> ExceptT Error m JWTClaims
-jwtClaims _ "" _ = return M.empty
-jwtClaims AppConfig{..} payload time = do
-  secret <- liftEither . maybeToRight JwtTokenMissing $ configJWKS
-  eitherClaims <-
-    lift . runExceptT $
-      JWT.verifyClaimsAt validation secret time =<< JWT.decodeCompact payload
-  liftEither . mapLeft jwtClaimsError $ claimsMap configJwtRoleClaimKey <$> eitherClaims
+  AppConfig -> Request -> UTCTime -> ExceptT Error m JWTClaims
+jwtClaims AppConfig{..} request time =
```

Comment:
We need some kind of naming convention for `PostgREST.Request`s and `Wai.Request`s. In `App.hs` it was `request` and `req`. Now `request` is the other one.

I wonder whether we should maybe not use `Request` and `Response` at all, given that those clash with wai like that.

What about `RestRequest` and `RestResponse`?

And then consistently use `request` for `Wai.Request` and `restRequest` for `PostgREST.RestRequest`.

Reply:
Hm, I think it would be cleanest to use `PostgREST.Request`and `PostgREST.Response` if possible in our codebase. The most impacted module is `PostgREST.App`, let me see if I can make it clearer there.

Reply:
`request` and `waiRequest` for names then?